src/extensions/list_status.rs

```diff
@@ -145,13 +145,14 @@ impl<T: Read + Write> Session<T> {
         data_items: &str,
     ) -> Result<ExtendedNames> {
         let reference = validate_str("LIST-STATUS", "reference", reference_name.unwrap_or(""))?;
-        self.run_command_and_read_response(&format!(
+        let lines = self.run_command_and_read_response(format!(
             "LIST {} {} RETURN (STATUS {})",
             &reference,
             mailbox_pattern.unwrap_or("\"\""),
             data_items
-        ))
-        .and_then(|lines| ExtendedNames::parse(lines, &mut self.unsolicited_responses_tx))
+        ))?;
+        let mut unsolicited_responses = self.all_unsolicited().collect();
```

Comment:
Why does this take + collect and then pass that to `ExtendedNames::parse` rather than pass `&mut self.unsolicited_responses` in like we do elsewhere?

Reply:
Because I get the following error:

```
rust-analyzer [E0616]: field `unsolicited_responses` of `Session` is private
```

which I do not understand because the fn is attached to `impl Session`. The work-around I found is to make the `unsolicited_responses` field `pub(crate)`, is that OK?

Reply:
Oh, yeah, totally fine to make it `pub(crate)` :+1:

Reply:
I _think_ you get it because privacy is about where the code that tries to access the field is located, not about the type it's on. For example, I can do:
```rust
trait MyTrait { fn foo(&self); }
impl MyTrait for std::collections::HashSet<bool> {
  fn foo(&self) {
    // but I don't get to access private fields of `self` here!
  }
}